src/main/proto/commands.proto

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+//
+// This file contains the protocol used to communicate between the Bazel client
+// and the server. At a high level clients may call the CommandServer.run rpc
+// to initiates a Bazel command and CommandServer.cancel to cancel an in-flight
+// command. CommandServer.ping may be used to check for server liveness without
+// executing any commands. See documentation of individual messages for more
+// details.
+syntax = "proto3";
+
+package remote_client;
+
+option java_package = "com.google.devtools.build.lib.remote.proxy";
+option java_multiple_files = true;
+option java_outer_classname = "CommandsProto";
+
+service Commands {
+  // Run a remote command and wait for completion.
+  rpc RunCommand (RunRequest) returns (stream RunResponse) {}
+}
+
+// The command to be executed remotely.
+message RunCommandParameters {
+  // An optional id to use to identify a build.
+  string build_request_id = 1;
+
+  // An optional id to use to identify an invocation.
+  string invocation_id = 2;
+
+  // An optional identifier of this command.
+  string id = 3;
+
+  // An optional tool name to pass to the remote server for logging.
+  string tool_name = 4;
+
+  // Whether to accept remotely cached action results.
+  bool accept_cached = 5;
+
+  // When set, this action results will not be cached remotely.
+  bool do_not_cache = 6;
+
+  // Input paths (files or directories) relative to exec root to include for
+  // command execution.
+  repeated string inputs = 7;
+
+  // Output files relative to exec root to download after command executes.
+  repeated string output_files = 8;
+
+  // Output directories relative to exec root to download after execution.
+  repeated string output_directories = 9;
+
+  // Command line elements to execute.
+  repeated string command = 10;
+
+  // Inputs to ignore, as regular expressions.
+  repeated string ignore_inputs = 11;
+
+  // Environment variables to pass through to remote execution.
+  map<string,string> environment_variables = 12;
+
+  // The platform to use for the remote execution.
+  map<string,string> platform = 13;
+
+  // Optional absolute path to save server logs for failed actions.
+  string server_logs_path = 14;
+
+  // If > 0, the maximum number of seconds to wait for remote action execution.
+  int32 execution_timeout = 15;
+
+  // When set, the proxy saves nthe full execution data for the executed action.
+  bool save_execution_data = 16;
+
+  // When set, the action will be run locally if it fails remotely.
+  bool local_fallback = 17;
+
+  // The working directory, relative to the exec root, for the command to run
+  // in. It must be a directory which exists in the input tree. If it is left
+  // empty, then the action is run in the exec root.
+  string working_directory = 18;
+}
+
+// Passed to CommandServer.run to initiate execution of a remote command.
+message RunRequest {
+  // Properties of the command to run.
+  RunCommandParameters command = 1;
```

Comment:
"command" is in two levels here. Is it possible to have a different name for this one? like command_parameters?

Reply:
I actually renamed the message to Comand instead, per Kaveh's suggestion, and named the internal one args